lib/galaxy/webapps/galaxy/api/visualizations.py

```diff
@@ -37,14 +44,66 @@
     BaseGalaxyAPIController,
     depends,
     DependsOnTrans,
+    IndexQueryTag,
     Router,
+    search_query_param,
+)
+from galaxy.webapps.galaxy.api.common import (
+    LimitQueryParam,
+    OffsetQueryParam,
 )
 from galaxy.webapps.galaxy.services.visualizations import VisualizationsService
 
 log = logging.getLogger(__name__)
 
 router = Router(tags=["visualizations"])
 
+DeletedQueryParam: bool = Query(
+    default=False, title="Display deleted", description="Whether to include deleted pages in the result."
+)
+
+UserIdQueryParam: Optional[DecodedDatabaseIdField] = Query(
+    default=None,
+    title="Encoded user ID to restrict query to, must be own id if not an admin user",
+)
+
+query_tags = [
+    IndexQueryTag("title", "The page's title."),
+    IndexQueryTag("slug", "The page's slug.", "s"),
+    IndexQueryTag("tag", "The page's tags.", "t"),
+    IndexQueryTag("user", "The page's owner's username.", "u"),
+]
```

Comment:
did you mean to say visualization instead of page ?

Reply:
Yes, thanks a lot for catching that and all the other comments!